packages/interactivity/src/proxies/context.ts

```diff
@@ -0,0 +1,69 @@
+// TODO: Use the proxy registry to avoid multiple proxies on the same object.
+
+// Store the context proxy and fallback for each object in the context.
+const contextObjectToProxy = new WeakMap();
+const contextProxyToObject = new WeakMap();
+const contextObjectToFallback = new WeakMap();
+
+const descriptor = Reflect.getOwnPropertyDescriptor;
+
+const contextHandlers: ProxyHandler< object > = {
+	get: ( target: object, k: string ) => {
+		const fallback = contextObjectToFallback.get( target );
+		// Always subscribe to prop changes in the current context.
+		const currentProp = target[ k ];
```

Comment:
Why not use `Reflect.get` here?

Reply:
It would be equivalent. 😄

In this case, we don't want to pass the context proxy as the receiver, so we would have to call `Reflect.get` like this:

```js
const currentProp = Reflect.get( target, k );
```

I can change it if you think it is more _correct_, though.

Reply:
> we don't want to pass the context proxy as the receiver

How so?

Reply:
Because it changes the receiver used inside the state proxy handlers to be the context proxy one.

You can test it with this example. It makes an object to return the receiver for any property.

```js
const obj = {}
const proxyA = new Proxy(
    obj,
    {
        get( _t, _k, receiver ) {
            return receiver; // We expect `receiver` to be `proxyA` here.
        }
    }
);

proxyA.prop === proxyA // true

const proxyB = new Proxy(
    proxyA, // It wraps proxyA.
    {
        get( t, k, receiver ) {
            // `t` is proxyA`, but we pass `proxyB` as the receiver
            return Reflect.get( t, k, receiver );
        }
    }
);


proxyB.prop === proxyA // false (should be true)
proxyB.prop === proxyB // true  (should be false)
```

Reply:
We're using proxy references to point to other related data, like reactive props, so we need to maintain the correct proxy reference (the state proxy one). 🙂

Reply:
That's interesting; thanks, David, for the explanation.